presto-iceberg/src/main/java/com/facebook/presto/iceberg/TableStatisticsMaker.java

```diff
@@ -121,8 +121,7 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle, Cons
         }
 
         TupleDomain<IcebergColumnHandle> intersection = constraint.getSummary()
-                .transform(IcebergColumnHandle.class::cast)
-                .intersect(tableHandle.getPredicate());
```

Comment:
Is removing this intersection fine or should we pass the valid predicate from layout?

Reply:
Thanks for the reminder. Yes, I think we should pass the valid predicate from layout or merge it into the `constraint` from outer this method. Will fix it.